### PR TITLE
Added tests for data factories

### PR DESCRIPTION
**Pull request by astrofrog**
This adds initial tests for the data factories. It's important not to include the full datasets in the repository as they are quite large, so this includes fake small arrays in the data instead of the original arrays, but otherwise preserves all the metadata. If we need larger data for some other tests, then we should use remote data (like astropy does with data.astropy.org).

All datasets included here (MIRI, SINFONI, MANGA) have been cleared to be public (even though we actually replace the data with small fake data anyway). I can include further datasets as they are authorized.

At the moment the test I've included is pretty rudimentary, as it just checks that the data loads with the correct factory, and that the data shape is correct. We could envisage including more sophisticated tests, but it's important to at least make sure most of the generic data factory code runs during the tests.

This also includes a bug fix that came to light with the tests (de752430b63fcb0a6bab1ad027204f382ed0fba2) and I've also removed the config.py file at the root of the repo (not sure why it's there?)

Fixes https://github.com/spacetelescope/cubeviz/issues/227

### CODE REVIEW

File: config.py [hunk 1]
```diff
@@ -1,6 +0,0 @@
-# Licensed under a 3-clause BSD style license - see LICENSE.rst
-
-import cubeviz.data_factories
-from cubeviz import layout
-from cubeviz.listener import CubevizManager
-from cubeviz import startup
```
Comment (brechmos-stsci) [inline config.py]: Just out of curiosity, why were these removed?  Is it a change in glue?  I don't remember why I/we added them but I thought they needed to be in here.

Comment (brechmos-stsci) [reply config.py]: Oh, just saw the question as to why it was there. :).  NVM.  Not 100% sure, I don't remember

Comment (astrofrog) [reply config.py]: config.py files are normally user configuration files - they shouldn't be in a package and aren't needed to run cubeviz.
